### DISCUSSION TITLE
$CURL_DEBUG docs

### DISCUSSION THREAD

**Discussion by gvanem** [curl/curl#10577] [Q&A] [answered]
I searched through all files under `docs/` and found no reference to the
environment variable `CURL_DEBUG` used in `curl_log.c`. Also nothing in `curl --manual`. 

I assume from reading the code it's on the form `set CURL_DEBUG=tcp,ssl,http3`. 
For a `-DDEBUGBUILD` and `curl --version`  would it be possible to list all flags and possibly add a `all` flag:
```diff
--- a/lib/curl_log.c 2023-02-10 15:26:24
+++ b/lib/curl_log.c 2023-02-21 10:31:31
@@ -189,15 +189,21 @@
     if(!tmp)
       return CURLE_OUT_OF_MEMORY;

-    token = strtok_r(tmp, ", ", &tok_buf);
-    while(token) {
-      for(i = 0; cf_types[i]; ++i) {
-        if(strcasecompare(token, cf_types[i]->name)) {
+    if(strcasecompare(tmp, "all")) {
+      for(i = 0; cf_types[i]; ++i)
           cf_types[i]->log_level = CURL_LOG_DEBUG;
-          break;
+    }
+    else {
+      token = strtok_r(tmp, ", ", &tok_buf);
+      while(token) {
+        for(i = 0; cf_types[i]; ++i) {
+          if(strcasecompare(token, cf_types[i]->name)) {
+            cf_types[i]->log_level = CURL_LOG_DEBUG;
+            break;
+          }
         }
+        token = strtok_r(NULL, ", ", &tok_buf);
       }
-      token = strtok_r(NULL, ", ", &tok_buf);
     }
     free(tmp);
   }
```

*BTW*. MSVC does not define a `__STDC_VERSION__ >= 199901L` to enable this log-feature.
But if a `-std:c11` is added, it's possible. Not sure if CMake or the .bat-files does that.

**Comment by bagder** [comment, marked as answer]
`CURL_DEBUG` is documented in `tests/README.md`

**Reply by gvanem** [reply]
Quote from `tests/README.md`:
> For a current list, one may search the sources for `struct Curl_cftype` definitions and find
> the names there. 

Rather hard to find IMHO.

**Reply by andy5995** [reply]
@gvanem I agree. I'm new to the project, but I would have expected to find something like that in the website docs (because the website docs are extensive and appear to me to be the primary source of documentation).

In fact, imho, the info from tests/README.md should be moved to the website, and the README should contain a link to the docs. What do you think?

**Reply by bagder** [reply]
Lots of docs on the website is simply converted from the documentation files in the source tree. It has the huge benefit that it then becomes available in both places without any duplicated work.

The `tests/READM.md` should probably get this treatment as well and be made available somewhere under https://curl.se/dev/ ... Any ideas on how/where?

**Reply by andy5995** [reply]
> The `tests/READM.md` should probably get this treatment as well and be made available somewhere under https://curl.se/dev/ ... Any ideas on how/where?

I started a change in https://github.com/curl/curl/pull/10691

My idea is to have the current [testcurl.html](https://curl.se/dev/testcurl.html) page contain the contents of tests/README.md

On the website, that's under "Dev related docs->Test curl"

Though as I think about, the file move I did in the PR probably isn't necessary.

**Reply by andy5995** [reply]
Website PR opened at https://github.com/curl/curl-www/pull/221
